client/extensions/woocommerce/components/product-search/index.js

```diff
@@ -6,64 +6,127 @@ import React, { Component } from 'react';
+		onChange: PropTypes.func.isRequired,
+		products: PropTypes.array,
 	};
 
-	state = {
-		query: '',
-	};
+	constructor( props ) {
+		super( props );
+		this.state = {
+			searchFilter: '',
+		};
+	}
 
-	handleSearch = query => {
+	componentDidMount() {
 		const { siteId } = this.props;
 
-		if ( trim( query ) === '' ) {
-			this.setState( { query: '' } );
-			return;
+		if ( siteId ) {
+			this.props.fetchProducts( siteId, { offset: 0, per_page: 50 } );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const { oldSiteId } = this.props;
+		const { newSiteId } = newProps;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchProducts( newSiteId, { offset: 0, per_page: 50 } );
 		}
+	}
+
+	getFilteredProducts() {
+		const { value, products } = this.props;
+		const { searchFilter } = this.state;
+
+		if ( 0 === searchFilter.length ) {
+			return products;
+		}
+
+		return (
+			products &&
+			products.filter( product => {
+				return (
+					productContainsString( product, searchFilter ) || isProductSelected( value, product.id )
+				);
+			} )
+		);
+	}
 
-		this.setState( { query } );
-		this.props.fetchProducts( siteId, { search: query } );
+	onSearch = searchFilter => {
+		this.setState( { searchFilter } );
 	};
 
-	handleSelect = product => {
-		// Clear the search field
-		this.setState( { query: '' } );
-		this.refs.searchCard.clear();
+	onProductCheckbox = productId => {
+		const { value } = this.props;
+		const selected = isProductSelected( value, productId );
+		const newValue = selected
+			? removeProductId( value, productId )
+			: addProductId( value, productId );
+		this.props.onChange( newValue );
+	};
 
-		// Pass products back to parent component
-		this.props.onSelect( product );
+	onProductRadio = productId => {
+		this.props.onChange( productId );
 	};
 
-	render() {
-		const { translate } = this.props;
+	renderSearch( searchFilter ) {
+		return <Search value={ searchFilter } onSearch={ this.onSearch } />;
+	}
+
+	renderList( singular ) {
+		const { value } = this.props;
+		const filteredProducts = this.getFilteredProducts() || [];
+		const renderFunc = product => {
+			const onChange = singular ? this.onProductRadio : this.onProductCheckbox;
+			return (
+				<ProductSearchRow
+					key={ product.id }
+					onChange={ onChange }
+					product={ product }
+					singular={ singular }
+					value={ value }
+				/>
+			);
+		};
+
+		return <div className="product-search__list">{ filteredProducts.map( renderFunc ) }</div>;
+	}
 
+	render() {
 		return (
 			<div className="product-search">
-				<SearchCard
-					ref="searchCard"
-					onSearch={ this.handleSearch }
-					delaySearch
-					delayTimeout={ 400 }
-					placeholder={ translate( 'Search products…' ) }
-				/>
-				<ProductSearchResults search={ this.state.query } onSelect={ this.handleSelect } />
+				{ this.renderSearch( this.state.searchFilter ) }
+				{ this.renderList( this.props.singular ) }
```

Comment:
Why not just destructure singular inside renderList instead of passing it? Same question for renderSearch too.